paper.md

```diff
@@ -0,0 +1,144 @@
+    affiliation: "1, 2"
+affiliations:
+ - name: Ames National Laboratory, Ames, IA, USA
+   index: 1
+ - name: Iowa State University, Ames, IA, USA
+   index: 2
+date: August 2022
+bibliography: paper.bib
+---
+
+# Summary
+
+This manuscript introduces CMinx, a program for generating
+application programming interface (API) documentation for CMake modules
+(code written in the CMake language). Since most of CMinx's intended audience
+is comprised of C/C++ developers, CMinx is designed to operate similar to
+Doxygen [@doxygen] (the *de facto* C/C++ API documentation tool).
+Specifically, developers annotate their CMake source with documentation
+comments (traditional CMake block comments, starting with an extra "`[`"
+character). The documentation comments, written in reST [@rest],
+describe to the reader how the functions, parameters, and variables should be
+used. Running CMinx on the annotated source code generates reST files containing
+the API documentation. The reST files can then be converted into static
+websites with tools such as Sphinx [@sphinx] or
+easily converted to another format via Pandoc [@pandoc].
+
+Unlike other solutions for documenting CMake modules, CMinx knows the CMake
+language's grammar. This enables CMinx to automatically extract function/macro
+signatures (even when functions are not documented). CMinx also integrates
+seamlessly into existing CMake build systems. CMinx's output is highly
+customizable and easily controlled via a YAML [@yaml] configuration file.
+CMinx has already proved to be an invaluable
+productivity tool in the authors' other projects, and, given that other
+scientific software projects also rely heavily on source code written in CMake,
+we anticipate CMinx will prove invaluable to many additional projects as well.
+
+# Statement of need
+
+The exascale-era of high-performance scientific computing arguably started when
+the Frontier supercomputer achieved a performance of more than 1.1 exaFLOPs
+(10$^{18}$ floating-point operations per second) on the High-Performance
+Linpack Benchmark [@frontier]. With the breakdown of Dennard scaling
+(*i.e.*, the power usage of a transistor is proportional to its area),
+achieving exascale performance is presently only possible by relying
+on heterogeneous hardware. Unfortunately, hardware heterogeneity significantly
+complicates the process of developing high-performance scientific software.
+For many high-performance software packages, these complications start in the
+build system.
+
+The process of building a software package written in a compiled language
+(*e.g.*, C or C++) has always been complicated. Over the years, various build
+system solutions have evolved to ease the process. Historically, there
+has been a propensity to treat each build system as a one-off use case. This is
+understandable since build systems have tended to be relatively small (*i.e.*,
+less than ~1K lines of code) and tightly coupled to the identity of the package.
```

Comment:
Unnecessary specification in parentheses. However, "identity" is too vague in my opinion:
```suggestion
understandable since build systems have tended to be relatively small
and tightly coupled to the structure and purpose of the package.
```